### PR TITLE
HBW-176 Add component for rendering created/deadline date

### PR DESCRIPTION
**Pull request by mxsmirnoff**

### CODE REVIEW

**Comment by mxsmirnoff** [comment]
image

**Comment by sugarfree1** [comment]
add screenshot with expired please

**Comment by mxsmirnoff** [comment]
Updated (expired now red)
image

File: hbw/app/javascript/packs/hbw/components/claiming/created.jsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/* eslint-disable implicit-arrow-linebreak */
+import React from 'react';
+import {
+  parseISO, isAfter, isSameYear,
+  differenceInMinutes, differenceInHours, differenceInDays, differenceInWeeks, differenceInYears
+} from 'date-fns';
+
+const HBWCreated = ({ env, created, due }) => {
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/claiming/created.jsx]: Extremely bad naming especially if this component can render created or due dates both. It's not even noun!

Comment (TimAle) [reply hbw/app/javascript/packs/hbw/components/claiming/created.jsx]: Also I would split this component to two: about due date formatting and about created date formatting, but I don't insist.

File: hbw/app/javascript/packs/hbw/init/date_localizer.js [hunk 2]
```diff
@@ -10,6 +12,22 @@ export const localize = (str, locale) => {
...
+);
+
+const localizeDateWithYear = locale => (
+  date => format(date, 'dd MMM YYYY', locales[locale])
+);
+
+export const dateLocalizer = locale => (
+  {
+    localizeForLocale:     localizeForLocale(locale),
+    localizeDateWithMonth: localizeDateWithMonth(locale),
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/init/date_localizer.js]: Let's add some consistency to this functions.

**Comment by mxsmirnoff** [comment]
image

File: hbw/app/javascript/packs/hbw/components/claiming/created_date.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/* eslint-disable implicit-arrow-linebreak */
+import {
+  isSameYear, differenceInMinutes, differenceInHours, differenceInDays
+} from 'date-fns';
+
+const HBWCreatedDate = ({ env, created }) => {
+  const { translator: t, localizer } = env;
+
+  const MINUTES_IN_DAY = 1440;
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/claiming/created_date.jsx]: Move duplicated code to HOC or module please

File: hbw/app/javascript/packs/hbw/components/claiming/created_date.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
+/* eslint-disable implicit-arrow-linebreak */
+import {
+  isSameYear, differenceInMinutes, differenceInHours, differenceInDays
+} from 'date-fns';
+
+const HBWCreatedDate = ({ env, created }) => {
```
Comment (TimAle) [inline hbw/app/javascript/packs/hbw/components/claiming/created_date.jsx]:
```suggestion
const HBWCreationDate = ({ env, date }) => {
```

**Pull request merged**